libyul/Object.cpp

```diff
@@ -62,13 +63,52 @@ string Object::toString(Dialect const* _dialect) const
 	return "object \"" + name.str() + "\" {\n" + indent(inner) + "\n}";
 }
 
-set<YulString> Object::dataNames() const
+set<YulString> Object::qualifiedDataNames() const
 {
-	set<YulString> names;
-	names.insert(name);
-	for (auto const& subObject: subIndexByName)
-		names.insert(subObject.first);
-	// The empty name is not valid
-	names.erase(YulString{});
-	return names;
+	set<YulString> qualifiedNames = name.str().empty() ? set<YulString>{} : set<YulString>{name};
+	for (shared_ptr<ObjectNode> const& subObjectNode: subObjects)
+	{
+		yulAssert(qualifiedNames.count(subObjectNode->name) == 0, "");
+		qualifiedNames.insert(subObjectNode->name);
+		if (auto const* subObject = dynamic_cast<Object const*>(subObjectNode.get()))
+			for (YulString const& subSubObj: subObject->qualifiedDataNames())
+				if (subObject->name != subSubObj)
+				{
+					yulAssert(qualifiedNames.count(YulString{subObject->name.str() + "." + subSubObj.str()}) == 0, "");
+					qualifiedNames.insert(YulString{subObject->name.str() + "." + subSubObj.str()});
+				}
+	}
+
+	yulAssert(qualifiedNames.count(YulString{}) == 0, "");
+	qualifiedNames.erase(YulString{});
+	return qualifiedNames;
+}
+
+vector<size_t> Object::pathToSubObject(YulString _qualifiedName) const
+{
+	yulAssert(_qualifiedName != name, "");
+	yulAssert(subIndexByName.count(name) == 0, "");
+
+	if (boost::algorithm::starts_with(_qualifiedName.str(), name.str() + "."))
+		_qualifiedName = YulString{_qualifiedName.str().substr(name.str().length() + 1)};
```

Comment:
Do we have a test that shows that `"currentObject."` is not a valid name?

Reply:
I added test, it shows that this is not a valid name

Reply:
In AsmAnalysis.cpp line 336
It is not a valid name for data object as it is not part of datanames

Reply:
`m_dataNames` are set to result of `qualifiedDataNames()`